Sphinx + Breathe + Doxygen docs + ReadTheDocs

oswinso opened the pull request:
## Description
This PR sets up the skeleton for documentation based off of Sphinx + Breathe + Doxygen.

## Associated Issue
Fixes #1475. 

## Steps to test
### Docs render properly.
1. Install new deps: `pip3 install sphinx breathe sphinx_rtd_theme`
2. Go to `build` folder
3. `ninja sphinx`

Expected result: There should be an `index.html` at `build/docs/docs_build/sphinx_docs` that shows the following:
![Screenshot_2020-08-17 Introduction — rj_robocup documentation](https://user-images.githubusercontent.com/4125463/90432607-cb60f000-e098-11ea-950d-f0a26021c642.png)

### ReadTheDocs works properly
1. Go to [https://rj-robocup.readthedocs.io/en/feat-ros2_doc](https://rj-robocup.readthedocs.io/en/feat-ros2_doc/)

Expected result: docs build and render properly.

oswinso committed:
Starting on documentation skeleton.

oswinso committed:
Added robobuzz logo

oswinso committed:
Change console language to sh for highlighting.

oswinso committed:
Filled out getting started and installation.

oswinso committed:
Add sphinx, rtd and breathe to requirements.

JNeiger commented on soccer/src/soccer/gameplay/gameplay_module.hpp:
```diff
@@ -39,7 +39,7 @@ namespace Gameplay {
  * calls into python, which does all of the high-level planning, resulting in
  * updated motion targets, etc for the robots.  The
- * GameplayModule then executes path planning for each OurRobot.
+ * GameplayModule then executes path planning for each ::OurRobot.
```
?

oswinso replied:
oh woops I was testing to see why breathe didn't link `OurRobot` so I was messing around. I'll revert this.

oswinso committed:
Revert change to GameplayModule

CollinAvidano commented:
What is the purpose of the src folder for docs? Also you all should probably set the depth of toc trees to 1. It makes it so its not just a glob of content in the nav bar. Try it out you will see what I mean. Also ideally the conf.py would be at the same level as the cmakelists.txt and doxyfile. Is there a particular reason this isnt the case?

CollinAvidano commented:
Also the top level index.rst appears to be an autogenerated one / duplicate of the one in src. If its not in use get rid of it.

oswinso committed:
Remove autogenerated index.rst

oswinso committed:
Created readthedocs.yml to specify location of conf.py
Added docs/requirements.txt for specifying documentation building dependencies.
Added docs/.gitignore to ignore build folders when building with sphinx-build.
Added hook to conf.py to run doxygen commands to generate XML for breathe.
Added Introduction (self) to toctree.
Added options to html_theme_options to never collapse the navigation.

oswinso committed:
Create CMake option default off for controlling whether documentation is built.

CollinAvidano reviewed (approved):
looks good to me.

CollinAvidano commented:
One thing that I didnt want to block the review on as its pretty minor was just maybe change the project name from rj-robocup to robocup-software for consistency with other repos. But done worry about it too much.

oswinso commented:
@CollinAvidano the problem with changing the project to `robocup-software` is that to other people, we're RoboJackets, not "robocup software"
